install/install.sh

```diff
@@ -193,8 +193,20 @@ detect_platform() {
     cygwin_nt*) platform="pc-windows-msvc";;
     # mingw is Git-Bash
     mingw*) platform="pc-windows-msvc" ;;
-    # use the statically compiled musl bins on linux to avoid linking issues.
-    linux) platform="unknown-linux-musl" ;;
+    linux)
+      # default to the statically compiled musl bins on linux
+      # to avoid linking issues.
+      platform="unknown-linux-musl"
+      if [ -f /etc/os-release ]; then
+	# on known-good distributions, use the dynamically compiled bins
+	distro=$(cat /etc/os-release | grep '^ID=' | sed -e 's/^ID=//')
```

Comment:
Just source it and work directly with `$DISTRIB_ID`?

Reply:
I pointed you towards neofetch for a good reason :upside_down_face: Here is how they do it:
```bash
            elif [[ -f /etc/os-release || \
                    -f /usr/lib/os-release || \
                    -f /etc/openwrt_release || \
                    -f /etc/lsb-release ]]; then

                # Source the os-release file
                for file in /etc/lsb-release /usr/lib/os-release \
                            /etc/os-release  /etc/openwrt_release; do
                    source "$file" && break
                done

                # Format the distro name.
                case $distro_shorthand in
                    on)   distro="${NAME:-${DISTRIB_ID}} ${VERSION_ID:-${DISTRIB_RELEASE}}" ;;
                    tiny) distro="${NAME:-${DISTRIB_ID:-${TAILS_PRODUCT_NAME}}}" ;;
                    off)  distro="${PRETTY_NAME:-${DISTRIB_DESCRIPTION}} ${UBUNTU_CODENAME}" ;;
                esac
```
Also, you're gonna need lowercase it because here is what I get on Linux Mint 19.2:
```sh
# from /etc/lsb-release
DISTRIB_ID=LinuxMint

# from /etc/os-release
ID=linuxmint
```

Reply:
And you can extend that (allow)list with Linux Mint because it is fairly popular Ubuntu derivate :wink:

Reply:
I think I prefer the old check https://github.com/starship/starship/commit/2e14d1af5a3b90aedab39180e0851e8f945b9f55

Reply:
```sh
if [ "$platform" = "linux" ] && ldd /bin/sh | grep -iq musl; then
  platform=unknown-linux-musl
fi
```

Reply:
so, something like the old check, but flipping it to detect glibc, would be ideal if we also add a check for the glibc version that works in a cross platform way, right?

Reply:
this seems to work. I'll rework the PR to use it
```
$ ldd --version | head -n1 | sed -e 's/ldd (GNU libc) //'
2.28
```

(first detecting if the string contains GNU libc, of course)

Reply:
You can simplify it by replacing `head` plus `sed` combo with awk:
```sh
ldd --version | awk 'NR==1 { print $NF }'
# or invoking /usr/lib/libc.so.6 directly
/usr/lib/libc.so.6 | awk 'NR==1 { print $NF }'
```

Reply:
Well, in that case you can use following checks:
```sh
ldd --version | awk '/GNU libc/ { print $NF }'
# or invoking /usr/lib/libc.so.6 directly
/usr/lib/libc.so.6 | awk '/GNU libc/ { gsub(/\.$/,""); print $NF }'
```

Reply:
> I can use `awk` if that's preferable

The sole reason why I prefer `awk` here is to minimize calls to external binaries.